components/configure-gradle-enterprise-maven-extension/build.gradle.kts

```diff
@@ -1,15 +1,17 @@
 plugins {
     id("java")
+    id("com.github.johnrengelman.shadow") version "8.1.1"
 }
 
 repositories {
     mavenCentral()
 }
 
 dependencies {
-    compileOnly("org.apache.maven:maven-core:3.9.7")
+    compileOnly("org.apache.maven:maven-core:3.6.3")  // intentionally compiling against an older version to preserve compatibility with older versions of Maven
     compileOnly("org.codehaus.plexus:plexus-component-annotations:2.2.0")
-    compileOnly("com.gradle:gradle-enterprise-maven-extension:1.18.4")
+    compileOnly("com.gradle:develocity-maven-extension:1.21.3")
```

Comment:
Should this be `testCompileOnly`or removed to prevent accidental direct usages of this instead of the adapters?

Reply:
Nice catch. There are no tests so it should be removed.

Reply:
Unfortunately, if this is removed, the project doesn't compile: https://ge.solutions-team.gradle.com/s/etr5srswtsk6y

It appears the develocity-maven-extension-adapters doesn't declare the develocity-maven-extension as a dependency.

We need classes in that library to compile because the ConfigureDevelocity and ConfigureGradleEnterprise listeners implement interfaces from that library.

Reply:
I guess that makes sense to avoid classpath conflicts in the adapters?

cc @pshevche

Reply:
> I guess that makes sense to avoid classpath conflicts in the adapters?

That would make sense. It's the same reason why we declare it as `compileOnly` instead of `implementation`: to avoid version conflicts with whatever version of the extension the user is actually using.

Reply:
ah that makes sense. good to keep it then.

Reply:
```suggestion
    compileOnly("com.gradle:develocity-maven-extension:1.21.4")
```